src/content/tutorials/build-a-qr-code-generator/index.md

```diff
@@ -0,0 +1,294 @@
+      body: JSON.stringify({ text: document.querySelector("#text").value })
+    })
+  }
+</script>
+`
+
+async function handleRequest(request) {
+  let response
+  if (request.method === 'POST') {
+    response = await generate(request)
+  } else {
+    response = new Response(landing, { headers: { 'Content-Type': 'text/html' } })
+  }
+  return response
+}
+```
+
+The `landing` variable, which is a static HTML string, sets up an `input` tag and a corresponding `button`, which calls the `generate` function. This function will make an HTTP `POST` request back to your serverless function, allowing you to see the corresponding QR code image data inside of your browser’s network inspector:
+
+![QR code generator working in the browser](./media/demo.png)
+
+With that, your serverless function is complete! The full version of the code looks like this:
+
+```javascript
+---
+filename: 'index.js'
+---
+const qr = require('qr-image')
+
+const generate = async request => {
+  const { text } = await request.json()
+  const headers = { 'Content-Type': 'image/png' }
+  const qr_png = qr.imageSync(text || 'https://workers.dev')
+  return new Response(qr_png, { headers })
+}
+
+const landing = `
+<h1>QR Generator</h1>
+<p>Click the below button to generate a new QR code. This will make a request to your serverless function.</p>
+<input type="text" id="text" value="https://workers.dev"></input>
+<button onclick='generate()'>Generate QR Code</button>
+<p>Check the "Network" tab in your browser's developer tools to see the generated QR code.</p>
+<script>
+  function generate() {
+    fetch(window.location.pathname, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ text: document.querySelector("#text").value })
+    })
+  }
+</script>
+`
+
+async function handleRequest(request) {
+  let response
+  if (request.method === 'POST') {
+    response = await generate(request)
+  } else {
+    response = new Response(landing, { headers: { 'Content-Type': 'text/html' } })
+  }
+  return response
+}
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+```
+
+## Publish
+
+And with that, you’re finished writing the code for the QR code serverless function, on Cloudflare Workers!
+
+Wrangler has built-in support for bundling, uploading, and releasing your Cloudflare Workers application. To do this, we’ll run `wrangler publish`, which will _build_ and _publish_ your code.
+
+```sh
+---
+header: Publishing your project
+---
+$ wrangler publish
+```
+
+## Resources
+
+In this tutorial, you built and published a serverless function to Cloudflare Workers for generating QR codes. If you’d like to see the full source code for this application, you can find it [on GitHub](https://github.com/signalnerve/workers-qr-code-generator).
+
+If you want to get started building your own projects, check out the quick-start templates we’ve provided in our [Template Gallery](/templates).
```

Comment:
Update to new ending seen in other new l18n tutorial (Template Gallery): 

https://github.com/cloudflare/workers-docs-engine/blob/5b0395e72f3c4d7d052fda7a4d8fc4d1b3946490/src/content/tutorials/localize-a-website/index.md#L281